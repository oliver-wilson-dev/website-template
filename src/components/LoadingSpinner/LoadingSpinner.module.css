@import "../../styles/colours.css";

.wrapper {
  display: flex;
  justify-content: center;
  align-items: center;
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  bottom: 0;
  background: $dark-grey;
  opacity: 0.5;

  visibility: hidden;
  opacity: 0;
  pointer-events: none;
  transition: all 0.25s ease-out;

  &Show {
    visibility: visible;
    opacity: 0.2;
    pointer-events: auto;
  }

  &Hide {
    transition-delay: 0.25s;
    visibility: hidden;
    opacity: 0;
    pointer-events: none;
  }
}

.loadingSpinner,
.loadingSpinner:after {
  border-radius: 50%;
  width: 10em;
  height: 10em;
}
.loadingSpinner {
  $spinnerWidth: 130px;
  $spinnerHeight: 130px;
  $spinnerBorder: 8px;
  width: $spinnerWidth;
  height: $spinnerHeight;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-grow: 1;
  font-size: 10px;
  border-top: $spinnerBorder solid var(--loading-spinner-border);
  border-right: $spinnerBorder solid var(--loading-spinner-border);
  border-bottom: $spinnerBorder solid var(--loading-spinner-border);
  border-left: $spinnerBorder solid var(--loading-spinner);
  transition: all 0.25s ease-out;
  transform: translateZ(0);

  animation: load8 1.1s infinite linear;

  position: fixed;
  top: 50%;
  top: calc(50% - calc($spinnerHeight / 2) - $spinnerBorder);
  left: calc(50% - calc($spinnerWidth / 2) - $spinnerBorder);

  &Show {
    transform: scale(1);
  }

  &Hide {
    transform: scale(0);
    animation: none;
    display: none;
  }
}

@keyframes load8 {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
