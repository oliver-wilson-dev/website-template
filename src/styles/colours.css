$background-grey: #e9ecef;
$dark-grey:  #333333;
$text-dark-grey:  #333333;
$text-light-grey:  #ffffff;
