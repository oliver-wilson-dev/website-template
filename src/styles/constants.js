export const LIGHT_THEME = Symbol('LIGHT_THEME');
export const DARK_THEME = Symbol('DARK_THEME');
