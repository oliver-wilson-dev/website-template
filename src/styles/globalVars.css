$max-content-width: 1000px;
$navigation-height: 50px;