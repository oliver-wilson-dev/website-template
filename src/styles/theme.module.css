@import "./colours.css";

.light {
    --background-colour: $background-grey;
    --background-colour-inverse: $dark-grey;
    --text-colour: $text-dark-grey;
    --text-colour-inverse: $text-light-grey;
    --side-menu-focus: $dark-grey;
}

.dark {
    --background-colour: $dark-grey;
    --background-colour-inverse: $background-grey;
    --text-colour: $text-light-grey;
    --text-colour-inverse: $text-dark-grey;
    --side-menu-focus: $background-grey;
}
