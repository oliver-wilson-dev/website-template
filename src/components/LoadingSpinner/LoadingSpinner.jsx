/*
  eslint-disable
  import/no-unresolved,
  jsx-a11y/click-events-have-key-events,
  jsx-a11y/no-static-element-interactions
*/
import React from 'react';
import cn from 'classnames';

import styles from './LoadingSpinner.module.css';

const onClick = (e) => {
  e.stopPropagation();
};

const LoadingSpinner = ({ show }) => (
  <>
    <div
      className={cn(styles.wrapper,
        {
          [styles.wrapperShow]: show,
          [styles.wrapperHide]: !show
        })}
      onClick={onClick}
    />
    <div className={cn(styles.loadingSpinner,
      {
        [styles.loadingSpinnerShow]: show,
        [styles.loadingSpinnerHide]: !show
      })}
    >
      <div />
      <div />
      <div />
      <div />
    </div>
  </>
);

export default LoadingSpinner;
