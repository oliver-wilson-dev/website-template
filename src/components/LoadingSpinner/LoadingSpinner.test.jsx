import React from 'react';
import { shallow } from 'enzyme';
import LoadingSpinner from './LoadingSpinner';
import styles from './LoadingSpinner.module.css';

const defaultProps = {
  show: true
};

const render = (props = {}) => shallow(<LoadingSpinner {...defaultProps} {...props} />);

describe('LoadingSpinner', () => {
  describe('when clicking on the overlay', () => {
    it('should stop the event propagation', () => {
      const stopPropagation = jest.fn();
      render().find(`.${styles.wrapper}`).simulate('click', { stopPropagation });

      expect(stopPropagation).toHaveBeenCalledWith();
    });
  });
  describe('when the component should show', () => {
    it('renders correctly', () => {
      expect(render()).toMatchSnapshot();
    });
  });

  describe('when the component should not show', () => {
    it('renders correctly', () => {
      expect(render()).toMatchSnapshot();
    });
  });
});
