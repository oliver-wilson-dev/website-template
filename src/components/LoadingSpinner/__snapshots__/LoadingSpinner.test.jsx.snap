// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner when the component should not show renders correctly 1`] = `
<Fragment>
  <div
    className="wrapper wrapperShow"
    onClick={[Function]}
  />
  <div
    className="loadingSpinner loadingSpinnerShow"
  >
    <div />
    <div />
    <div />
    <div />
  </div>
</Fragment>
`;

exports[`LoadingSpinner when the component should show renders correctly 1`] = `
<Fragment>
  <div
    className="wrapper wrapperShow"
    onClick={[Function]}
  />
  <div
    className="loadingSpinner loadingSpinnerShow"
  >
    <div />
    <div />
    <div />
    <div />
  </div>
</Fragment>
`;
